flow/layers/texture_layer.cc

```diff
@@ -32,6 +32,7 @@ void TextureLayer::Preroll(PrerollContext* context, const SkMatrix& matrix) {
 
 void TextureLayer::Paint(PaintContext& context) const {
   TRACE_EVENT0("flutter", "TextureLayer::Paint");
+  // FML_DCHECK(needs_painting(context));
```

Comment:
I need to double check this. We never used to DCHECK on TextureLayer, but I'm not sure if it was just an oversight or if this causes a problem. I need to make sure I have a practical test case to verify that this won't cause a problem before I add this.

Reply:
(Certainly we have the unit tests, but I'm not sure how comprehensive they are.)

Reply:
Don't think I've seen `layer->Paint()` call anywhere that wasn't within a `layer->needs_painting()` condition (except for the tests).

Reply:
I tried enabling it and discovered that the TextureLayer "make sure that painting an empty texture layer dies" test was not actually prepared to have the assertion fail on it. I'll fix the test and add the DCHECK in...

Reply:
Indeed it was the tests that were the problem. The tests should not be fixed and the DCHECK is now live in the code (debug mode only obviously).